cdc/model/mounter.go

```diff
@@ -49,13 +51,23 @@ func (e *PolymorphicEvent) RegionID() uint64 {
 	return e.RawKV.RegionID
 }
 
+// IsResolved returns true if the event is resolved.
+func (e *PolymorphicEvent) IsResolved() bool {
+	return e.RawKV != nil && e.RawKV.OpType == OpTypeResolved
+}
+
+// IsBatchResolved returns true if the event is batch resolved event.
+func (e *PolymorphicEvent) IsBatchResolved() bool {
+	return e.IsResolved() && e.Batch
+}
+
 // ComparePolymorphicEvents compares two events by CRTs, Resolved, StartTs, Delete/Put order.
 // It returns true if and only if i should precede j.
 func ComparePolymorphicEvents(i, j *PolymorphicEvent) bool {
 	if i.CRTs == j.CRTs {
-		if i.RawKV.OpType == OpTypeResolved {
+		if i.IsResolved() {
 			return false
-		} else if j.RawKV.OpType == OpTypeResolved {
+		} else if j.IsResolved() {
```

Comment:
It looks that this `else` is unnecessary.

Reply:
Else is necessary, otherwise this function will return `false` when `i.IsResolved()==false` and `j.IsResolved()==true`.